test/link.test.js

```diff
@@ -121,80 +121,9 @@ after(function (cb) {
 
 test('put link', function (t) {
     this.client.ln(this.obj, this.key, function (err, res) {
-        t.ifError(err);
-        t.checkResponse(res, 204);
-        t.end();
-    });
-});
-
-
-test('link to public', function (t) {
-    this.client.ln(this.pubKey, this.key, function (err, res) {
-        t.ifError(err);
-        t.checkResponse(res, 204);
-        t.end();
-    });
-});
-
-
-test('put link parent ENOEXIST', function (t) {
-    var k = this.key + '/' + uuid.v4();
-    this.client.ln(this.obj, k, function (err, res) {
-        t.ok(err);
-        t.equal(err.name, 'DirectoryDoesNotExistError');
-        t.checkResponse(res, 404);
-        t.end();
-    });
-});
-
-
-test('put parent not directory', function (t) {
-    var k = this.obj + '/' + uuid.v4();
-    this.client.ln(this.obj, k, function (err, res) {
-        t.ok(err);
-        t.equal(err.name, 'ParentNotDirectoryError');
-        t.checkResponse(res, 400);
-        t.end();
-    });
-});
-
-
-test('put source not found', function (t) {
-    var src = this.dir + '/' + uuid.v4();
-    this.client.ln(src, this.key, function (err, res) {
-        t.ok(err);
-        t.equal(err.name, 'SourceObjectNotFoundError');
-        t.checkResponse(res, 404);
-        t.end();
-    });
-});
-
-
-test('put if-match', function (t) {
-    var self = this;
-    var opts = {
-        headers: {
-            'if-match': self.etag
-        }
-    };
-    this.client.ln(this.obj, this.key, opts, function (err, res) {
-        t.ifError(err);
-        t.checkResponse(res, 204);
-        t.end();
-    });
-});
-
-
-test('put if-match fail', function (t) {
-    var opts = {
-        headers: {
-            'if-match': uuid.v4()
-        }
-    };
-    this.client.ln(this.obj, this.key, opts, function (err, res) {
-        t.ok(err);
-        t.equal(err.name, 'PreconditionFailedError');
-        t.checkResponse(res, 412);
+        t.ok(err && err.code === 'SnaplinksDisabledError',
+            'Expected SnapLinks to be disabled');
+        t.checkResponse(res, 403);
```

Comment:
We use the existing error on purpose because clients will already have had to (if they are correct) support this. If we want a different code, let me know which one you want and I'll change it though we'll probably want to check all clients support that.

Reply:
Yup fair. I don't see node-manta ever having been updated to handle this gracefully. I haven't tested what it looks like. Have you tried `mln ...` against your muskie with your patch? What does it look like (I'm curious)?

Reply:
Yep, testing notes in the ticket, but this looks like:

```
failhammer:~ JOSHSNAP$ mln ~~/stor/hello.txt ~~/stor/hello2.txt
mln: SnaplinksDisabledError: Manta v2 does not support SnapLinks.
failhammer:~ JOSHSNAP$
failhammer:~ JOSHNOSNAP$ mln ~~/stor/hello.txt ~~/stor/hello2.txt
mln: SnaplinksDisabledError: Manta v2 does not support SnapLinks.
failhammer:~ JOSHNOSNAP$
```

where JOSHNOSNAP is a user in the accountsSnaplinksDisabled array and JOSHSNAP is a user not in the array.